sd/etcd/registrar.go

```diff
@@ -0,0 +1,54 @@
+package etcd
+
+import (
+	etcd "github.com/coreos/etcd/client"
+	"github.com/go-kit/kit/log"
+)
+
+// Registrar registers service instance liveness information to etcd.
+type Registrar struct {
+	client  Client
+	service Service
+	logger  log.Logger
+}
+
+// Service holds the key, value and instance identifying data you
+// want to publish to etcd.
+type Service struct {
+	Key           string // discovery key, example: /myorganization/myplatform/
+	Value         string // service name value, example: addscv
+	Data          []byte // instance data to store for discovery, example: 10.0.2.10:80
```

Comment:
Given users are constructing this in their own code, does it make sense to have this be a `string` and do the conversion internally?